wsd/Admin.cpp

```diff
@@ -685,6 +691,28 @@ void Admin::pollingThread()
         LOG_TRC("Admin poll for " << timeout);
         poll(timeout); // continue with ms for admin, settings etc.
     }
+
+    if (!COOLWSD::IndirectionServerEnabled)
+        return;
+
+    _model.sendShutdownReceivedMsg();
+
+    constexpr size_t sleepMs = 1000;
+    const auto migrationMsgTimeout =
+        COOLWSD::getConfigValue<int>("indirection_endpoint.migration_timeout_secs", 180) * 1000;
+    size_t count = migrationMsgTimeout / sleepMs;
+    for (size_t i = 0; i < count; ++i)
+    {
+        poll(std::chrono::seconds(0));
+        if (_closeMonitor)
+        {
+            LOG_INF("Received closemonitor message");
+            break;
+        }
+
+        LOG_DBG("Waiting for migration to complete before closing the monitor");
+        std::this_thread::sleep_for(std::chrono::milliseconds(sleepMs));
+    }
```

Comment:
It would be best to log a warning when the wait fails. To do that easily, you can check for `_closeMonitor` after the loop. If it's not set, log a warning.

Reply:
> It would be best to log a warning when the wait fails. To do that easily, you can check for `_closeMonitor` after the loop. If it's not set, log a warning.

`LOG_WRN("Timeout waiting for 'closemonitor' message");` 
Is this warning fine or you are talking about other thing ?